### ISSUE TITLE
URISyntaxException when setting route.url with trailing slash

### ISSUE THREAD

**Issue by msparer** [Netflix/zuul#173]
Using version 1.0.28, the problem occurs if you define a route like this (using spring-boot in this example):

```
zuul:
  routes:
    foos:
      path: /foos/**
      url: http://localhost:8080/
```
The trailing slash of the ```url``` always gets prepended to the delegation request. E.g.

```curl -X GET http://zuulhost/foos/1```

results in a ```GET //1``` to ```url``` (instead of ```GET /1```). This particular example does work though, a call to root doesn't however:

```curl -X GET http://zuulhost/foo/``` --> ```URISyntaxException``` as the request is ```GET //```

The current workaround of course is to define ```url``` without trailing slash, then everything works as expected:

```
zuul:
  routes:
    foos:
      path: /foos/**
      url: http://localhost:8080
```

tl;dr:
```ZuulRoute``` should take care of trailing slashes in ```url``` on delegation

**Comment by lleiguo** [comment]
On Oct 9, 2015, at 6:52 AM, Michael Sparer <notifications@github.com<mailto:notifications@github.com>> wrote:


Using version 1.0.28, the problem occurs if you define a route like this (using spring-boot in this example):

zuul:
  routes:
    foos:
      path: /foos/**
      url: http://localhost:8080/


The trailing slash of the url always gets prepended to the delegation request. E.g.

curl -X GET http://zuulhost/foos/1

results in a GET //1 to url (instead of GET /1). This particular example does work though, a call to root doesn't however:

curl -X GET http://zuulhost/foo/ --> URISyntaxException as the request is GET //

The current workaround of course is to define url without trailing slash, then everything works as expected:

zuul:
  routes:
    foos:
      path: /foos/**
      url: http://localhost:8080


tl;dr:
ZuulRoute should take care of trailing slashes in url on delegation

**Comment by spencergibb** [comment]
@msparer `zuul.routes` is provided by [spring-cloud-netflix](https://github.com/spring-cloud/spring-cloud-netflix), not by zuul proper. You should open an issue [there](https://github.com/spring-cloud/spring-cloud-netflix/issues) instead.

**Comment by msparer** [comment]
@spencergibb oh sorry, missed that. Opened [the issue there](https://github.com/spring-cloud/spring-cloud-netflix/issues/583)

**Comment by NiteshKant** [comment]
Thanks @spencergibb. I will close this issue then?

**Comment by spencergibb** [comment]
yes please.
